### PR TITLE
chore(connlib): pass to client new fields

### LINKED ISSUE #4885: Send all info from portal onUpdateResources to clients
- [ ] addressDescription
- [ ] Site information

### PR DESCRIPTION
**Pull request by conectado**
Fixes #4885

### CODE REVIEW

File: rust/connlib/shared/src/messages/client.rs [hunk 1]
```diff
@@ -31,6 +35,33 @@ pub struct ResourceDescriptionCidr {
...
+
+    pub address_description: String,
+    pub gateway_groups: Vec<GatewayGroup>,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct GatewayGroup {
+    pub name: String,
+    pub id: SiteId,
+}
```
Comment (ReactorScram) [inline rust/connlib/shared/src/messages/client.rs:47]: Is a Gateway group different from a Site?

**Review by ReactorScram — APPROVED** [review]
LGTM

File: rust/connlib/shared/src/messages/client.rs (hunk 1 above)
Comment (conectado) [reply rust/connlib/shared/src/messages/client.rs:47]: I think it's just another name for site cc @AndrewDryga

Comment (thomaseizinger) [reply rust/connlib/shared/src/messages/client.rs:47]: Can we call it `Site` then and rename the field using serde? Perhaps with a comment that the API should be changed?

**Comment by jamilbk** [comment]
@conectado Was this tested with all the clients to make sure adding the fields doesn't cause serialization errors on the other side of the FFI? Or was that not implemented yet?

**Comment by thomaseizinger** [comment]
> @conectado Was this tested with all the clients to make sure adding the fields doesn't cause serialization errors on the other side of the FFI? Or was that not implemented yet?

This PR only added fields to the messages received from the portal. No changes to FFI.

**Pull request merged**